### PR TITLE
Interconnect Template

### PR DESCRIPTION
**Pull request by jlittle-sourced**
This is the complete interconnect template.  There is no option to create a partner interconnect only dedicated which is why I named the example appropriately

### CODE REVIEW

File: community/cloud-foundation/templates/interconnect/examples/interconnect_dedicated.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Example of the Interconnect Attachment template usage.
```
Comment (sourced-pavel) [inline community/cloud-foundation/templates/interconnect/examples/interconnect_dedicated.yaml]: `Interconnect Attachment` or `Interconnect`?

File: community/cloud-foundation/templates/interconnect/README.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
+# Interconnect Attachment
```
Comment (sourced-pavel) [inline community/cloud-foundation/templates/interconnect/README.md]: Attachment?

File: community/cloud-foundation/templates/interconnect/README.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+This template creates an Interconnect
+
+## Prerequisites
+
+- Install [gcloud](https://cloud.google.com/sdk)
+- Create a [GCP project, set up billing, enable requisite APIs](../project/README.md)
+- Create a [network](../network/README.md)
+- Create a [interconnect] (../interconnect/README.md)
+- Create a [cloud_router] (../cloud_router/README.md)
+- Grant the [compute.networkAdmin](https://cloud.google.com/compute/docs/access/iam) i
```
Comment (sourced-pavel) [inline community/cloud-foundation/templates/interconnect/README.md]: possibly orphaned whitespace + `i` at the end

File: community/cloud-foundation/templates/interconnect/README.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    cd community/cloud-foundation
+```
+
+3. Copy the example DM config to be used as a model for the deployment; in this case, [examples/interconnect_dedicated.yaml](examples/interconnect_dedicated.yaml):
+
+```shell
+    cp templates/interconnect/examples/interconnect_dedicated.yaml my_interconnect_dedicated.yaml
+```
+
+4. Change the values in the config file to match your specific GCP setup (for
```
Comment (sourced-pavel) [inline community/cloud-foundation/templates/interconnect/README.md]: trailing whitespace

File: community/cloud-foundation/templates/interconnect/README.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    git clone https://github.com/GoogleCloudPlatform/deploymentmanager-samples
+```
+
+2. Go to the [community/cloud-foundation](../../) directory:
+
+```shell
+    cd community/cloud-foundation
+```
+
+3. Copy the example DM config to be used as a model for the deployment; in this case, [examples/interconnect_dedicated.yaml](examples/interconnect_dedicated.yaml):
```
Comment (sourced-pavel) [inline community/cloud-foundation/templates/interconnect/README.md]: >80 characters

File: community/cloud-foundation/templates/interconnect/README.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+```
+
+4. Change the values in the config file to match your specific GCP setup (for 
+   properties, refer to the schema files listed above):
+
+```shell
+    vim my_interconnect_dedicated.yaml  # <== change values to match your GCP setup
+```
+
+5. Create your deployment (replace \<YOUR\_DEPLOYMENT\_NAME\> with the relevant
```
Comment (sourced-pavel) [inline community/cloud-foundation/templates/interconnect/README.md]: trailing whitespace

File: community/cloud-foundation/templates/interconnect/examples/interconnect_dedicated.yaml (hunk 1 above)
Comment (jlittle-sourced) [reply community/cloud-foundation/templates/interconnect/examples/interconnect_dedicated.yaml]: dammit forgot to use -i when I grepped for it thought I had them all

Comment (jlittle-sourced) [reply community/cloud-foundation/templates/interconnect/examples/interconnect_dedicated.yaml]: fixed

File: community/cloud-foundation/templates/interconnect/README.md (hunk 2 above)
Comment (jlittle-sourced) [reply community/cloud-foundation/templates/interconnect/README.md]: fixed

File: community/cloud-foundation/templates/interconnect/README.md (hunk 3 above)
Comment (jlittle-sourced) [reply community/cloud-foundation/templates/interconnect/README.md]: done

File: community/cloud-foundation/templates/interconnect/README.md (hunk 4 above)
Comment (jlittle-sourced) [reply community/cloud-foundation/templates/interconnect/README.md]: done

File: community/cloud-foundation/templates/interconnect/README.md (hunk 5 above)
Comment (jlittle-sourced) [reply community/cloud-foundation/templates/interconnect/README.md]: done

File: community/cloud-foundation/templates/interconnect/README.md (hunk 6 above)
Comment (jlittle-sourced) [reply community/cloud-foundation/templates/interconnect/README.md]: done

**Pull request closed without merging**